validator/sawtooth_validator/gossip/permission_verifier.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+import logging
+# pylint: disable=import-error,no-name-in-module
+# needed for google.protobuf import
+from google.protobuf.message import DecodeError
+
+from sawtooth_validator.protobuf import client_pb2
+from sawtooth_validator.protobuf.batch_pb2 import BatchHeader
+from sawtooth_validator.protobuf.validator_pb2 import Message
+from sawtooth_validator.networking.dispatch import HandlerResult
+from sawtooth_validator.networking.dispatch import HandlerStatus
+from sawtooth_validator.networking.dispatch import Handler
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def check_batches_signing_keys(batches, allowed_pubkeys):
+    for batch in batches:
+        if not check_batch_signing_key(batch, allowed_pubkeys):
+            return False
+    return True
+
+
+def check_batch_signing_key(batch, allowed_pubkeys):
+    # validate batch signature
+    if allowed_pubkeys is not None:
+        header = BatchHeader()
+        header.ParseFromString(batch.header)
+        if header.signer_pubkey not in allowed_pubkeys:
+            LOGGER.info("Batch was signed by an unauthorized signing key %s",
+                        header.signer_pubkey)
+            return False
+    return True
+
+
+class BatchListPermissionVerifier(Handler):
+    def __init__(self, config_view_factory, current_root_func):
+        self.config_view_factory = config_view_factory
+        self.current_root_func = current_root_func
+
+    def handle(self, connection_id, message_content):
+        response_proto = client_pb2.ClientBatchSubmitResponse
+        allowed_pubkeys = None
+        try:
+            state_root = self.current_root_func()
+            config_view = \
+                self.config_view_factory.create_config_view(state_root)
+            allowed_pubkeys = config_view.get_setting(
+                "sawtooth.validator.allowed_signing_key")
+        except AttributeError:
+            LOGGER.debug("Chain head not yet set")
+
+        def make_response(out_status):
```

Comment:
What is the reasoning behind _not_ defining helper functions within other functions? 

In this case `make_response` only makes sense in the context of the `handle` method, as it only abstracts away a common pattern used in `handle`, and references a variable in `handle`'s scope.